src/factory/TaikoDaoFactory.sol

```diff
@@ -0,0 +1,323 @@
+                OptimisticTokenVotingPluginSetup.InstallationParameters(
+                    votingSettings,
+                    existingTokenSettings,
+                    unusedMintSettings,
+                    settings.taikoL1ContractAddress,
+                    settings.taikoBridgeAddress,
+                    settings.minStdProposalDelay,
+                    stdProposer,
+                    emergencyProposer
+                )
+            );
+        }
+
+        (address plugin, IPluginSetup.PreparedSetupData memory preparedSetupData) = settings
+            .pluginSetupProcessor
+            .prepareInstallation(
+            address(dao),
+            PluginSetupProcessor.PrepareInstallationParams(
+                PluginSetupRef(PluginRepo.Tag(1, 1), PluginRepo(pluginRepo)), settingsData
+            )
+        );
+
+        return (OptimisticTokenVotingPlugin(plugin), pluginRepo, preparedSetupData);
+    }
+
+    function deployPublicKeyRegistry() internal returns (PublicKeyRegistry) {
+        return new PublicKeyRegistry();
+    }
+
+    function applyPluginInstallation(
+        DAO dao,
+        address plugin,
+        PluginRepo pluginRepo,
+        IPluginSetup.PreparedSetupData memory preparedSetupData
+    ) internal {
+        settings.pluginSetupProcessor.applyInstallation(
+            address(dao),
+            PluginSetupProcessor.ApplyInstallationParams(
+                PluginSetupRef(PluginRepo.Tag(1, 1), pluginRepo),
+                plugin,
+                preparedSetupData.permissions,
+                hashHelpers(preparedSetupData.helpers)
+            )
+        );
+    }
+
+    function grantApplyInstallationPermissions(DAO dao) internal {
+        // The PSP can manage permissions on the new DAO
+        dao.grant(address(dao), address(settings.pluginSetupProcessor), dao.ROOT_PERMISSION_ID());
+
+        // This factory can call applyInstallation() on the PSP
+        dao.grant(
+            address(settings.pluginSetupProcessor),
+            address(this),
+            settings.pluginSetupProcessor.APPLY_INSTALLATION_PERMISSION_ID()
+        );
+    }
+
+    function revokeApplyInstallationPermissions(DAO dao) internal {
+        // Revoking the permission for the factory to call applyInstallation() on the PSP
+        dao.revoke(
+            address(settings.pluginSetupProcessor),
+            address(this),
+            settings.pluginSetupProcessor.APPLY_INSTALLATION_PERMISSION_ID()
+        );
+
+        // Revoke the PSP permission to manage permissions on the new DAO
+        dao.revoke(address(dao), address(settings.pluginSetupProcessor), dao.ROOT_PERMISSION_ID());
+    }
+
+    function revokeOwnerPermission(DAO dao) internal {
+        dao.revoke(address(dao), address(this), dao.ROOT_PERMISSION_ID());
+    }
+
+    // Getters
+
+    function getSettings() public view returns (DeploymentSettings memory) {
```

Comment:
`getSettings()` and `getDeployment()` could be removed by having both variables as public.
There's no upgradeability needed here, not custom logic and these functions are not being used inside the contract. So I think having them public would be better.

Reply:
My concern here is that if we mix 15 settings variables and 8 artifact addresses, this would result in a bloated ABI and it would also make it difficult for consumers to see what is what.

By having two distinct functions, it's clear what is a setting and what is an artifact, and the ABI becomes trivial